### PR TITLE
Dev QoL changes

### PR DESCRIPTION
**Pull request by petorsong**
Add some dev quality of life changes - mainly docker/vscode integration and linting
more context here: https://www.notion.so/uwblueprintexecs/Docker-workflow-192d476ab8524123b0a057fa00e48440

[Notion task](https://www.notion.so/uwblueprintexecs/9224caeb21ce4350b8a1dc378d393323?v=43f44ac04d734394b7f7edc636c523b4)

## Changes
- add vscode docker integration
  - on your local vscode, need to first install `remote development` extension
- add linter (prettier + eslint)
- bump typescript version to latest and manually convert js files to ts(x)
- add backend switch to not have hardcoded localhost

## Testing
- tested that opening vscode in container works
- tested that linter is correctly formatting on save
  - e.g. change a string to use single quotes then save -> should switch to double quotes

### CODE REVIEW

**Review by rickymao — DISMISSED** [review]

File: src/index.tsx [hunk 1]
```diff
@@ -22,3 +23,9 @@ ReactDOM.render(
 // unregister() to register() below. Note this comes with some pitfalls.
 // Learn more about service workers: https://bit.ly/CRA-PWA
 serviceWorker.unregister();
+
+Sentry.init({
+  dsn: process.env.REACT_APP_DSN,
```
Comment (petorsong) [inline src/index.tsx:28]: sentry config is here

File: src/graphql/apollo/client.ts [hunk 2]
```diff
@@ -1,12 +1,10 @@
 import ApolloClient from "apollo-boost";
 import defaults from "./defaults";
-// import resolvers from "../resolvers";
 
 const client = new ApolloClient({
-  uri: "http://localhost:4000/",
+  uri: process.env.REACT_APP_BACKEND_HOST || "http://localhost:4000/",
```
Comment (petorsong) [inline src/graphql/apollo/client.ts:6]: backend config for deployed hosts is here

File: .gitignore [hunk 3]
```diff
@@ -8,6 +8,9 @@
 # testing
 /coverage
 
+#dotenv
+.env
+
 # production
 /build
```
Comment (aaronabraham311) [inline .gitignore:16]: Do we need to add the builder file for the gitignore?

**Review by aaronabraham311 — COMMENTED** [review]
Looks good asides from the question about the builder file. It ran on my local machine without a problem.  
Edit: never mind, the builder was already included.

**Review by aaronabraham311 — APPROVED** [review]

**Review by rickymao — APPROVED** [review]
LGTM

**Pull request merged**